modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/PartitionMetaManager.java

```diff
@@ -0,0 +1,168 @@
+        metas.put(groupPartitionId, partitionMeta);
+    }
+
+    /**
+     * Reads the partition {@link PartitionMeta meta} from the partition file or creates a new one.
+     *
+     * <p>If it creates a new one, it writes the meta to the file.
+     *
+     * @param checkpointId Checkpoint ID.
+     * @param groupPartitionId Partition of the group.
+     * @param pageMemory Page memory.
+     * @param filePageStore Partition file page store.
+     */
+    public PartitionMeta readOrCreateMeta(
+            @Nullable UUID checkpointId,
+            GroupPartitionId groupPartitionId,
+            PersistentPageMemory pageMemory,
+            FilePageStore filePageStore
+    ) throws IgniteInternalCheckedException {
+        ByteBuffer buffer = allocateBuffer(pageSize);
+
+        long bufferAddr = bufferAddress(buffer);
+
+        long partitionMetaPageId = pageMemory.partitionMetaPageId(groupPartitionId.getGroupId(), groupPartitionId.getPartitionId());
+
+        try {
+            if (filePageStore.size() > filePageStore.headerSize()) {
+                // Reads the partition meta.
+                if (filePageStore.pages() == 0) {
+                    filePageStore.pages(1);
+                }
+
+                boolean read = filePageStore.read(partitionMetaPageId, buffer, false);
+
+                assert read : filePageStore.filePath();
+
+                return createMeta(ioRegistry.resolve(bufferAddr), bufferAddr, checkpointId);
+            } else {
+                // Creates and writes a partition meta.
+                PartitionMetaIo partitionMetaIo = PartitionMetaIo.VERSIONS.latest();
+                partitionMetaIo.initNewPage(bufferAddr, partitionMetaPageId, pageSize);
+
+                // Because we will now write this page.
+                partitionMetaIo.setPageCount(bufferAddr, 1);
+
+                filePageStore.pages(1);
+
+                filePageStore.write(partitionMetaPageId, buffer.rewind(), true);
+
+                filePageStore.sync();
+
+                return createMeta(partitionMetaIo, bufferAddr, checkpointId);
+            }
+        } finally {
+            freeBuffer(buffer);
+        }
+    }
+
+    /**
+     * Writes the partition meta to the buffer.
+     *
+     * @param groupPartitionId Partition of the group.
+     * @param pageMemory Page memory.
+     * @param partitionMeta Snapshot of the partition meta.
+     * @param writeToBuffer Direct byte buffer to write partition meta.
+     */
+    public void writeMetaToBuffer(
+            GroupPartitionId groupPartitionId,
+            PersistentPageMemory pageMemory,
+            PartitionMetaSnapshot partitionMeta,
+            ByteBuffer writeToBuffer
+    ) {
+        assert writeToBuffer.remaining() == pageSize : writeToBuffer.remaining();
+
+        long partitionMetaPageId = pageMemory.partitionMetaPageId(groupPartitionId.getGroupId(), groupPartitionId.getPartitionId());
```

Comment:
So, the only reason of passing page memory instance is to construct meta page id? Are there helper methods for it somewhere? I'd ask you to remove page memory parameter in such places.

Reply:
Sounds reasonable.